src/main/java/com/company/skyproeducationspring/utils/IntListImpl.java

```diff
@@ -0,0 +1,214 @@
+package com.company.skyproeducationspring.utils;
+
+import java.util.Arrays;
+
+public class IntListImpl implements IntList {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    /**
+     * The array buffer into which the elements of the IntListImpl are stored.
+     */
+    private int[] data;
+
+    /**
+     * The size of the StringListImpl.
+     */
+    private int listSize = 0;
+
+    private int listCapacity;
+
+    public IntListImpl() {
+        listCapacity = DEFAULT_CAPACITY;
+        this.data = new int[listCapacity];
+    }
+
+    public IntListImpl(int capacity) {
+        listCapacity = capacity;
+
+        if (capacity >= 0) {
+            data = new int[listCapacity];
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: " + capacity);
+        }
+    }
+
+    public IntListImpl(int[] inputData) {
+        data = inputData;
+        listSize = inputData.length;
+        listCapacity = inputData.length;
+    }
+
+    @Override
+    public int add(int item) {
+
+        if (listSize == listCapacity) {
+            expand();
+        }
+
+        data[listSize] = item;
+        ++listSize;
+
+        return data[listSize - 1];
+    }
+
+    @Override
+    public int add(int index, int item) {
+        if (index < 0 || index >= listCapacity) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+
+        data[index] = item;
+
+        return data[index];
+    }
+
+    @Override
+    public int set(int index, int item) {
+        if (index < 0 || index >= listCapacity) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+
+        data[index] = item;
+
+        return data[index];
+    }
+
+    @Override
+    public int remove(int index) {
+        int string = get(index);
+
+        System.arraycopy(data, index + 1, data, index, listSize);
+        listSize--;
```

Comment:
также есть смысл обнулить последний дублирующийся элемент после сдвига, чтобы сборщик мусора мог его обнулить
```suggestion
        data[listSize - 1] = null;
        listSize--;
```

Reply:
Нельзя заNULLить `'int'`

Reply:
да, все верно, будет работать только в случае, если все переделаем на Integer (править необязательно)